tests/Aspire.Cli.Tests/Commands/ExecCommandTests.cs

```diff
@@ -75,12 +75,30 @@ public async Task ExecCommand_WhenProjectFileDoesNotExist_ReturnsFailedToFindPro
         var provider = services.BuildServiceProvider();
 
         var command = provider.GetRequiredService<RootCommand>();
-        var result = command.Parse("exec");
+        var result = command.Parse("exec --resource api");
 
         var exitCode = await result.InvokeAsync().WaitAsync(CliTestConstants.DefaultTimeout);
         Assert.Equal(ExitCodeConstants.FailedToFindProject, exitCode);
     }
 
+    [Fact]
+    public async Task ExecCommand_WhenTargetResourceNotSpecified_ReturnsInvalidCommand()
+    {
+        using var workspace = TemporaryWorkspace.Create(_outputHelper);
+        var services = CliTestHelper.CreateServiceCollection(workspace, _outputHelper, options =>
+        {
+            options.ProjectLocatorFactory = _ => new TestProjectLocator();
+        });
+        var provider = services.BuildServiceProvider();
+
+        var command = provider.GetRequiredService<RootCommand>();
+        var result = command.Parse("exec --project test.csproj echo hello");
+
+        // should not take long and fail fastly
+        var exitCode = await result.InvokeAsync().WaitAsync(TimeSpan.FromSeconds(3));
```

Comment:
This is very flaky. Asserting something happened quickly in a CI unit test won't be reliable because CI machines can randomly be slow.

Would it be better to look at the output instead?